website/content/docs/rfcs/external-keys.mdx

```diff
@@ -0,0 +1,214 @@
+    the name of the library (if PKCS#11), and all credential information.
+
+  In particular, this means that there is no central repository (within a
+  namespace) of credentials, making rotation harder.
+
+Further, while less restrictive, any namespace can use any `kms_library` or
+seal type; a namespace admin has full control over policies and thus a global
+operator cannot restrict types.
+
+## User-facing Description
+
+We suggest the following design for external keys:
+
+1. The `external_keys` stanzas take an optional `namespaces = []` parameter,
+   taking items of the form `uuid:<ns-uuid>`, `id:<ns-accessor>`, or
+   `path:<ns-path>`, to identify namespaces that this library is allowed to be
+   used in. This prevents cross-tenant library usage problems by preventing them
+   from using the underlying library. An empty `namespaces` parameter implicitly
+   allows the root namespace, a non-empty list must explicitly include the root
+   namespace (e.g., via `id:root`) to keep it allowed besides any other listed
+   namespaces.
+
+   - In the future, when support for pluginized KMS integrations are added, we
+     can use an `external_keys "binary" { ... }` stanza to limit access to
+     specific external libraries or hosts that they run on.
+
+   - The `name` of this stanza gets registered as the library type; it must
+     not conflict with any existing type and `pkcs11` will not be a valid
+     type, instead requiring a named configuration. E.g., `gcpckms`, `thales`,
+     `entrust`, and `securosys` might all be valid `types` for the endpoints
+     below.
+
+   For example, to register the SoftHSM PKCS#11 library as library type
+   `"softhsm"` and allow its usage in two namespaces:
+
+   ```hcl
+   external_keys "pkcs11" {
+     name       = "softhsm"
+     library    = "/usr/lib/softhsm/libsofthsm2.so"
+     namespaces = ["path:foo/bar", "uuid:c0de1570-688e-41ad-ad97-1395fe30cbbd"]
+   }
+   ```
+
+2. Within `sys/namespaces/<:path>/*`, we add a new option, `external_key_types`,
+   which limits the allowed HSM and KMS types within this namespace and all
+   children. This allows both the root operator to restrict immediate children,
+   but also for future tenant operators to restrict their own child namespaces.
+   This reflects the hierarchical nature of access (that a parent namespace
+   operator can create a policy that effectively grants them full access to a
+   child namespace).
+
+3. Within the `sys/external-keys` space, we implement the following APIs:
+
+   - `configs/<:config-name>`, to configure KMS or HSM _access_ information,
+     such as type (above) or credentials. This gives a single point of rotation
+     for any given key using this provider.
+
+     A configuration may be inherited from a direct parent namespace via the
+     special `inherits = "<:config-name>"` key. This inheritance mechanism
+     is chosen over a top-level `inheritable = true` flag to avoid potential
```

Comment:
```suggestion
     is chosen over a top-level `inherited = true` flag to avoid potential
```
`inheritable = true` would be in the parent namespace.